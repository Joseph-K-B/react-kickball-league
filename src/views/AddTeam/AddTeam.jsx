import { useState } from "react"
import { useHistory } from "react-router-dom"

function AddTeam() {
    const [name, setName] = useState('');
    const [city, setCity] = useState('');
    const [state, setState] = useState('');
    const histor = useHistory();

    const handleSubmit = async (e) => {
        e.preventDefault();

        // const res = await createTeam({name, sity, state})
        // history.pushState(`/teams/${res[0].id}`)
    }

    return(
        <>
        <fieldset>
            <legend>Add Team</legend>
            <form>
                <label htmlFor='name'>Name:</label>
                <input id='name' type="text" value={name} onChange={({target}) => setName(target.value)}/>

                <label htmlFor='city'>City:</label>
                <input id='city' type="text" value={city} onChange={({target}) => setCity(target.value)}/>

                <label htmlFor='state'>State:</label>
                <input id='state' type="text" value={state} onChange={({target}) => setState(target.value)}/>

                <button aria-label='add team' type="submit" onSubmit={handleSubmit}>Add</button>
            </form>
        </fieldset>
        </>
    )
}

export default AddTeam;