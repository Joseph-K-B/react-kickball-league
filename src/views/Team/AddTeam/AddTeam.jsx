import { useState } from "react"
import { useHistory } from "react-router-dom"
import TeamForm from "../../../components/TeamForm";
import { createTeam } from "../../../services/teams";

function AddTeam() {
    const [name, setName] = useState('');
    const [city, setCity] = useState('');
    const [state, setState] = useState('');
    const history = useHistory();

    const handleAddSubmit = async (e) => {
        e.preventDefault();
        const res = await createTeam({name, city, state})
        history.push(`/teams/${res[0].id}`)
    }

    return(
        <>
        <fieldset>
            <legend>Add Team</legend>
            <TeamForm 
                name={name}
                city={city}
                state={state}
                handleSubmit={handleAddSubmit}
                setName={setName}
                setCity={setCity}
                setState={setState}
            />
        </fieldset>
        </>
    )
}

export default AddTeam;