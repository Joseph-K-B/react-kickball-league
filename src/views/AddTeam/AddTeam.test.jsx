import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import { MemoryRouter, Router, Route } from 'react-router-dom';
import { createMemoryHistory } from "history";
import userEvent from '@testing-library/user-event'
import AddTeam from './AddTeam';

const mockTeam = {
        id: 9,
        created_at: '',
        name: 'redirect',
        city: 'random',
        state: 'Vatican City',
        players: []
}

const server = setupServer(
    rest.get('url', (req, res, ctx) => {
        return res(
            ctx.json(mockTeam)
        )
    }),
    rest.post('url', (req, res, ctx) => {
        return res (
            ctx.json([
                mockTeam
            ])  
        );

    })
);

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

it('renders form to add team and redirects to team details', async () => {
    const history = createMemoryHistory();
    history.push('/teams/new');

    render(
        <Router>
            <Route>
            <AddTeam />
            </Route>
        </Router>
    );

    screen.getByText('Add a Team');

    const nameField = screen.getByLabelText(/Name:/);
    const cityField = screen.getByLabelText(/City:/);
    const stateField = screen.getByLabelText(/State:/);
    const submitBtn = screen.getByRole('button', {cname: 'add team' })

    userEvent.type(nameField);
    userEvent.type(cityField);
    userEvent.type(stateField);
    userEvent.click(submitBtn);

})