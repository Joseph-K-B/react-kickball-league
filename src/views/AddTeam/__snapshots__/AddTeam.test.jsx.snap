// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form to add team and redirects to team details 1`] = `undefined`;
